src/Block/SharedBlockBlockService.php

```diff
@@ -121,13 +118,11 @@ public function getMetadata(): MetadataInterface
 
     public function load(BlockInterface $block): void
     {
-        $sharedBlock = $block->getSetting('blockId', null);
+        $blockId = $block->getSetting('blockId', null);
 
-        if (\is_int($sharedBlock)) {
-            $sharedBlock = $this->blockManager->findOneBy(['id' => $sharedBlock]);
+        if (!$blockId instanceof PageBlockInterface) {
```

Comment:
This is weird having a setting blockId which is not an id but the object. Should we rename it ?

Reply:
We can maybe rename the variable, but not the key of the array. otherwise we will break everything stored on the db of users migrating from 3 to 4.

it is called blockId because on the db is stored as id, but I agree it could been easier to simply call id block.

Reply:
```
$this->blockManager->findOneBy(['id' => $blockId])
```
can return null.
Should we throw an exception in that case ?

Reply:
> We can maybe rename the variable, but not the key of the array. otherwise we will break everything stored on the db of users migrating from 3 to 4.
> 
> it is called blockId because on the db is stored as id, but I agree it could been easier to simply call id block.

Then we can keep it like this indeed. We might just need to handle the case where we don't find the block in database.

Reply:
I guess not, I think is better to leave null.

Use case:

1. create a page
2. create a block pointing to that page
3. remove the page
4. edit the block, you dont want to get an exception, block wont be automatically removed but instead it will be empty

I guess thats how it should work.